sklearn/ensemble/_iforest.py

```diff
@@ -484,6 +486,65 @@ def _more_tags(self):
         }
 
 
+# Lookup table used below in _average_path_length() for small samples
+_average_path_length_small = np.array(
+    (
+        0.0,
+        0.0,
+        1.0,
+        1.6666666666666666667,
+        2.1666666666666666667,
+        2.5666666666666666667,
+        2.9000000000000000000,
+        3.1857142857142857143,
+        3.4357142857142857143,
+        3.6579365079365079365,
+        3.8579365079365079365,
+        4.0397546897546897547,
+        4.2064213564213564214,
+        4.3602675102675102675,
+        4.5031246531246531247,
+        4.6364579864579864580,
+        4.7614579864579864580,
+        4.8791050452815158698,
+        4.9902161563926269809,
+        5.0954793142873638230,
+        5.1954793142873638230,
+        5.2907174095254590611,
+        5.3816265004345499702,
+        5.4685830221736804049,
+        5.5519163555070137383,
+        5.6319163555070137383,
+        5.7088394324300906613,
+        5.7829135065041647354,
+        5.8543420779327361640,
+        5.9233075951741154743,
+        5.9899742618407821410,
+        6.0544903908730402055,
+        6.1169903908730402055,
+        6.1775964514791008116,
+        6.2364199808908655175,
+        6.2935628380337226603,
+        6.3491183935892782159,
+        6.4031724476433322699,
+        6.4558040265907006910,
+        6.5070860778727519730,
+        6.5570860778727519730,
+        6.6058665656776300218,
+        6.6534856132966776409,
+        6.6999972412036543850,
+        6.7454517866581998396,
+        6.7898962311026442840,
+        6.8333744919722095014,
+        6.8759276834615712036,
+        6.9175943501282378702,
+        6.9584106766588501151,
+        6.9984106766588501151,
+        7.0376263629333599190,
+    )
+)
```

Comment:
Where is this table coming from? Could you generate it instead of hard-coding it?

Reply:
Having a lookup table here is useful because it's fast. The corresponding value could be calculated each time from the definition of the harmonic numbers, but especially for the later values, this isn't very efficient. It is significantly slower than the asymptotic expansion, which can be used for larger values, where it becomes sufficiently accurate. 
BTW, I'm not really happy about the formatting, but that's what the utility _Black_ generated.

Reply:
Yes, I agree with your point but I think having it definition be explicitly analytic than having it hard-coded is better for readers and for maintenance.

Probably we can find a common ground combining efficiency and explicitness. 

What is the snippet that you have used to generate this table?

Reply:
Thanks for your suggestion. The values were generated using SymPy to get correct results. One could do something like `np.sum(2/np.arange(2, n))`, but as you can easily see this would be much slower and may include roundoff errors. I understand your point about readability, so how about we keep the list but document it explicitly with definition of the harmonic numbers and a code snippet explaining how the list can be generated? This way it will be clear to anyone reading the code where the list came from and we also keep the benefit of superior performance and accuracy.

Reply:
It would be great including the snippet generating this list. :+1:

Reply:
Alright, I added to documentation to the code. The main line to generate the list is `[Sum(2/k, (k, 2, max(1, i))).evalf(22).round(19) for i in range(52)]`, it is mentioned just above the list. Let me know what you think of it, I also added an explanation how the calculation is split for large and small samples.